### ISSUE TITLE
False-positive "superfluous-parens" after "in" keyword

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#4959]
### Bug description

```python
"""Bug."""
for x in ((0, slice(0)),):
    print(x[0])  # 0
    print(x[1])  # slice
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module bug
bug.py:2:0: C0325: Unnecessary parens after 'in' keyword (superfluous-parens)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.9.6 (default, Jan  1 1970, 00:00:00) 
[GCC 7.5.0]
```

### OS / Environment

OpenSUSE LEAP 15.2

**Comment by Pierre-Sassoulas** [comment]
This look like it's fixed in main, it will be available in 2.11.0.
